src/averageAges.js

```diff
@@ -55,7 +73,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const onlySon = people.filter((person) =>
+    onlyWithSon
+      ? person.sex === 'm'
+        && people.some((mom) => {
+          if (person.mother === mom.name) {
+            person['motherBorn'] = mom.born;
+
+            return true;
+          }
+        })
+      : people.some((mom) => {
+        if (person.mother === mom.name) {
+          person['motherBorn'] = mom.born;
+
+          return true;
+        }
+      })
+  );
+
+  const lifespan = onlySon.reduce(
+    (sum, age) => sum + (age.born - age.motherBorn),
+    0
+  );
+
+  const averageAge = +(lifespan / onlySon.length).toFixed(2);
+
+  return averageAge;
```

Comment:
```suggestion
  return lifespan / onlySon.length;
```
looks like redundat to fixed